### ISSUE TITLE
Add bidirectional edge mode

### ISSUE THREAD

**Issue by nrabinowitz** [uber/h3#133]
It occurred to me that it would be trivial to add a bidirectional edge mode (I would have preferred "directed" and "undirected", but we already have "unidirectional"), just by taking the unidirectional edge of the lower H3 index in a given pair of cells. I don't have use cases in mind yet, but it seems like a useful primitive, and it's simple to implement on top of the unidirectional functions. 

If we had a more efficient version of `getH3UnidirectionalEdgeBoundary` this might allow an even simpler optimization for `h3SetToLinkedGeo` than vertex mode.

**Comment by dfellis** [comment]
Yep! Making the smaller index the first one is a good solution. Validating these bidirectional edges should also mark an inverted form as invalid.

A more efficient uni/bidirectional edge also would need to understand icosahedron face IJK rotations to determine the proper direction number when dealing with hexagons on the "edge" (which will depend on the res 0 parent owning icosahedron, not the icosahedron it is physically on).

Given those complexities I am not convinced a more efficient algorithm would be significantly more efficient.

**Comment by nrabinowitz** [comment]
This would now be called `edge`
